### PR TITLE
new Paper-only event `player opens sign`

### PR DESCRIPTION
**Pull request by pobab**

### CODE REVIEW

File: paper/src/main/java/com/denizenscript/denizen/paper/events/PlayerOpenSignScriptEvent.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    //
+    // @Plugin Paper
+    //
+    // @Group Paper
+    //
+    // @Location true
+    //
+    // @Cancellable true
+    //
+    // @Triggers When player open sign.
```
Comment (mcmonkey4eva) [inline paper/src/main/java/com/denizenscript/denizen/paper/events/PlayerOpenSignScriptEvent.java]: that ain't English

File: paper/src/main/java/com/denizenscript/denizen/paper/events/PlayerOpenSignScriptEvent.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    //
+    // @Context
+    // <context.side> returns an ElementTag of side the sign was clicked.
```
Comment (mcmonkey4eva) [inline paper/src/main/java/com/denizenscript/denizen/paper/events/PlayerOpenSignScriptEvent.java]: grammar issues here as well

File: paper/src/main/java/com/denizenscript/denizen/paper/events/PlayerOpenSignScriptEvent.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        };
+    }
+
+    @EventHandler
+    public void playerOpenSignEvent(PlayerOpenSignEvent event) {
+        location = new LocationTag(event.getSign().getLocation());
+        this.event = event;
+        fire(event);
+    }
+}
```
Comment (mcmonkey4eva) [inline paper/src/main/java/com/denizenscript/denizen/paper/events/PlayerOpenSignScriptEvent.java]: NL@EoF

**Comment by mcmonkey4eva** [comment]
What's the reason for this event? Did somebody request it? What reason would you use this event that other events don't cover?

**Comment by pobab** [comment]
> What's the reason for this event? Did somebody request it? What reason would you use this event that other events don't cover?

oh. i forgot to mention. this event requested from here https://discord.com/channels/315163488085475337/1136954189646401557

**Pull request merged**